docs/section5/Rule5-24.md

```diff
@@ -0,0 +1,93 @@
+
+      - Else if window-wall-ratio is between 20% to 30%, get baseline vertical fenestration construction U-factor requirement: `if ( wwr_semiheated > 20.0% ) AND ( wwr_semiheated <= 30.0% ): target_u_factor_semiheated = data_lookup(table_G3_4, climate_zone, "SEMIHEATED", "VERTICAL GLAZING", "20.1%-30.0%", "ASSEMBLY MAX. U")`
+
+      - Else, window-wall-ratio is between 30% to 40%, get baseline vertical fenestration construction U-factor requirement: `if ( wwr_semiheated > 30.0% ) AND ( wwr_semiheated <= 40.0% ): target_u_factor_semiheated = data_lookup(table_G3_4, climate_zone, "SEMIHEATED", "VERTICAL GLAZING", "30.1%-40.0%", "ASSEMBLY MAX. U")`
+
+  - For each zone in building: `for zone_b in building_b...zones:`
+
+    - For each surface in zone: `for surface_b in zone_b.surfaces:`
+
+      - Check if surface is above-grade wall with subsurface and is regulated: `if ( get_opaque_surface_type(surface_b) == "ABOVE-GRADE WALL" ) AND ( surface_b.subsurfaces ) AND ( scc_dictionary_b[surface_b.id] != "UNREGULATED" ):`
+
+        - For each subsurface in wall: `for subsurface_b in surface_b.subsurfaces:`
+
+          - Check if subsurface is door and glazed area is more than 50% of the total door area, or subsurface is not door: `if (( subsurface_b.classification == "DOOR" ) AND ( subsurface_b.glazed_area > subsurface_b.opaque_area )) OR ( subsurface_b.classification != "DOOR" ):`
+
+            **Rule Assertion:**
+
+            - Case 1: For each subsurface that is vertical fenestration, if wall is exterior mixed type and the baseline requirements for residential and non-residential type U-factor for different window-wall-ratio are different: `if manual_review_flag AND ( scc_dictionary_b[surface_b] == "EXTERIOR MIXED" ): CAUTION and raise_warning "MANUAL REVIEW IS REQUESTED TO VERIFY VERTICAL FENESTRATION MEETS U-FACTOR REQUIREMENT AS PER TABLE G3.4."`
+
+            - Case 2: Else if wall is exterior mixed type and vertical fenestration U-factor matches Table G3.4 requirement: `else if ( scc_dictionary_b[surface_b] == "EXTERIOR MIXED" ) AND ( subsurface_b.u_factor == target_u_factor_mixed ): PASS`
+
+            - Case 3: Else if wall is exterior residential type and vertical fenestration U-factor matches Table G3.4 requirement: `else if ( scc_dictionary_b[surface_b] == "EXTERIOR RESIDENTIAL" ) AND ( subsurface_b.u_factor == target_u_factor_res ): PASS`
+
+            - Case 4: Else if wall is exterior non-residential type and vertical fenestration U-factor matches Table G3.4 requirement: `else if ( scc_dictionary_b[surface_b] == "EXTERIOR NON-RESIDENTIAL" ) AND ( subsurface_b.u_factor == target_u_factor_nonres ): PASS`
+
+            - Case 5: Else if wall is semi-exterior type and skylight U-factor matches Table G3.4 requirement: `else if ( scc_dictionary_b[surface_b] == "SEMI-EXTERIOR" ) AND ( subsurface_b.u_factor == target_u_factor_semiheated ): PASS`
+
+            - Case 6: Else: `else: FAIL`
```

Comment:
My only comment is that it RDS produces fail/pass outcome for each surface as opposed to one per RDS

Reply:
I think that is the case with all RDS. as discussed, in such cases we will simply report the number pass/fail/caution for all surfaces and the reviewer will decide the outcome for the rule.

Reply:
Updated. Added Rule Assertion - RMR.

Reply:
@xingcx I love this approach for Rule Assertion- Component and Rule Assertion- RMR. we should do this in all applicable RDS going forward!